libhb/decavcodec.c

```diff
@@ -1152,6 +1153,13 @@ int reinit_video_filters(hb_work_private_t * pv)
     enum AVPixelFormat pix_fmt;
     enum AVColorRange  color_range;
 
+#if HB_PROJECT_FEATURE_NVENC
```

Comment:
Unfortunately you can't avoid having a filter here. In some rare cases a video can change size, and HandBrake output it's always in limited range, so it needs to convert the range here if the source is full range.

Reply:
I wonder if we have the same issue for QSV or if it's handling it better upstream somewhere. It looks like the code below is doing something similar.

Reply:
From what I can see there isn't any hw scale and color range filter in your latest commit.

Reply:
@galad87 

>In some rare cases a video can change size, and HandBrake output it's always in limited range, so it needs to convert the range here if the source is full range.

Regarding the input resolution change - what's the expected behavior here?
Do we reconfigure encoder to new resolution or do we scale frames between decoder and encoder to avoid encoder reconfigure?

Reply:
Scale the frames, the scale (and color range) is configured here in this function.

Reply:
@galad87  @sr55 

>Unfortunately you can't avoid having a filter here


Scaling can be done on GPU with the help of `scale_npp` or `scale_cuda` ffmpeg filters. For range conversion probably another ffmpeg GPU-accelerated filter has to be developed.

However, enabling CUDA and NPP filters requires some changes in ffmpeg build configuration. Is this acceptable?